artiq/coredevice/comm_kernel.py

```diff
@@ -688,7 +688,7 @@ def read_exception_string():
 
         try:
             python_exn = python_exn_type(
-                nested_exceptions[-1][1].format(*nested_exceptions[0][2]))
+                nested_exceptions[-1][1].format(*nested_exceptions[0][2]) if nested_exceptions[-1][7] else nested_exceptions[-1][1])
```

Comment:
If you are already going through this code, would you mind documenting what's going on with the -1/0 indices here? This seems very non-obvious at a glance.

Reply:
Isn't it a mistake? Shouldn't the params be tied to the last exception?

Reply:
Possibly; I just haven't had a look at what's happening there in any detail (hence the mention of test cases :P).

Reply:
Well it was a mistake. On a nested exception case it was linking messages to the exception in wrong way. Added test for this